Added VOLUME entry to Docker Image (#117)
exasol/data-science-sandbox: doc/user_guide/docker/secure_configuration_storage.md

# Secure Configuration Storage

Most of the examples in the Exasol AI-Lab (XAI) require a connection to an Exasol database. Additionally some show cases will interact with cloud services which require additional configuration items, URL endpoints, credentials, etc.

> To keep all these configuration options accross multiple sessions, XAI offers a _Secure Configuration Storage_ (SCS).

Reviewer: ```suggestion
To keep all these configuration options accross multiple sessions and secure, the Exasol AI-Lab offers a _Secure Configuration Storage_ (SCS).
```

The Secure Configuration Storage is based on [coleifer/sqlcipher3](https://github.com/coleifer/sqlcipher3) which uses an encrypted version of an SQLite database. The database is stored in an ordinary, yet encrypted, file and allows to store credentials and other configuration strings in a simple key-value style.

Access to the SCS is encapsulated by a Python library to simplify usage as much as possible.

> When using the same file then you can reuse all your configurations and credentials in each AI-Lab session while still staying secure as the file is encrypted with a master password that is only known to you.

Reviewer: ```suggestion
When using the same file, you can reuse all your configurations and credentials in each AI-Lab session while still staying secure as the file is encrypted with a master password that is only known to you.
```

> When the file initially does not exist then AI-Lab will

Reviewer: ```suggestion
When the file initially does not exist then the AI-Lab will
```

> * Ask you for this master password

Reviewer: ```suggestion
* Ask you for a new master password
```

* Create the file
* Encrypt the file with the master password

In each future session AI-Lab will
* Ask you again for the master password

> * Use it to unlock and access the access the configuration strings stored in previous sessions.

Reviewer: ```suggestion
* Use it to unlock and access the configuration strings stored in previous sessions.
```
